### PR TITLE
Exposes Newton collision pipeline configuration through NewtonCfg

### PR DESCRIPTION
**Pull request by ooctipus**
# Description

Expose Newton's `CollisionPipeline` parameters through a new `NewtonCollisionPipelineCfg` configclass, allowing users to customize broad phase, contact limits, and hydroelastic mode via `NewtonCfg.collision_cfg`.

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab_newton/isaaclab_newton/physics/newton_manager.py
Comment (AntoineRichard) [inline source/isaaclab_newton/isaaclab_newton/physics/newton_manager.py:1]: I'm wondering if the config resolution for the contacts could be resolved outside of the manager. Maybe using something similar to the Kamino solver cfg? https://github.com/isaac-sim/IsaacLab/pull/5214

**Review by AntoineRichard — CHANGES_REQUESTED** [review]

**Comment by mzamoramora-nvidia** [comment]
Clean, minimal PR that exposes Newton's `CollisionPipeline` parameters through `NewtonCfg.collision_cfg`. Well-scoped — it does one thing (pipeline config pass-through) and does it correctly. The mutual exclusion guard with `use_mujoco_contacts` is a good safety check.

---

## Specific feedback

### `HydroelasticSDFCfg` — missing parameters

`HydroelasticSDFCfg` exposes 6 of the 13 parameters on Newton's `HydroelasticSDF.Config`. The missing ones are:

| Newton `HydroelasticSDF.Config` field | Exposed? |
|---|---|
| `reduce_contacts` | Yes |
| `buffer_fraction` | Yes |
| `normal_matching` | Yes |
| `anchor_contact` | Yes |
| `margin_contact_area` | Yes |
| `output_contact_surface` | Yes |
| **`pre_prune_contacts`** | **No** |
| **`moment_matching`** | **No** |
| **`buffer_mult_broad`** | **No** |
| **`buffer_mult_iso`** | **No** |
| **`buffer_mult_contact`** | **No** |
| **`contact_buffer_fraction`** | **No** |
| **`grid_size`** | **No** |

Of these, `moment_matching` and the `buffer_mult_*` multipliers are practically important — `moment_matching` affects friction accuracy in contact reduction, and `buffer_mult_*` are the primary knobs users reach for when they hit overflow warnings. Suggest adding at least `moment_matching`, `buffer_mult_broad`, `buffer_mult_iso`, `buffer_mult_contact`, and `grid_size`.

Since `HydroelasticSDF.Config` has all 13 params, exposing only 6 means users have to fall back to constructing `HydroelasticSDF.Config` directly for advanced tuning, bypassing the config system.

### `_initialize_contacts` — `to_dict()` may include `None` values

```python
cfg_dict = cls._collision_cfg.to_dict()
hydro_cfg = cfg_dict.pop("sdf_hydroelastic_config", None)
if hydro_cfg:
    cfg_dict["sdf_hydroelastic_config"] = HydroelasticSDF.Config(**hydro_cfg)
cls._collision_pipeline = CollisionPipeline(cls._model, **cfg_dict)
```

If `to_dict()` includes fields with `None` values (e.g. `rigid_contact_max=None`, `soft_contact_max=None`, `requires_grad=None`), passing them as `**cfg_dict` works because `CollisionPipeline.__init__` accepts `None` for those params. But worth verifying that `to_dict()` doesn't serialize `None` fields differently than expected (some configclass implementations skip `None` fields, others include them).

### `tolerance` field on `MJWarpSolverCfg` — unrelated to this PR?

The PR adds a `tolerance: float = 1e-6` field to `MJWarpSolverCfg`. This doesn't seem related to collision pipeline configuration. If it's intentional, it should be in a separate commit or mentioned in the PR description. If it slipped in, consider removing to keep the PR focused.

### No tests

The PR checklist shows tests unchecked. Even a basic test that creates a `NewtonCollisionPipelineCfg`, sets `collision_cfg` on `NewtonCfg`, and verifies the `CollisionPipeline` is created with the right params would catch regressions. The `to_dict()` -> `CollisionPipeline(**cfg_dict)` path is the most important thing to test.

**Comment by mzamoramora-nvidia** [comment]
Hi @ooctipus, I have some more feedback on this PR. Maybe you already planned to address it on separete PR but it's worth noting a potential issue on the setup of SDF/Hydroelastic shapes.

## Context

PR #5219 correctly exposes `CollisionPipeline` parameters through `NewtonCollisionPipelineCfg` and `HydroelasticSDFCfg`. However, there is a gap in the end-to-end workflow: **nothing in this PR ensures that shapes have SDF data or the HYDROELASTIC flag set**.

## The Problem

For hydroelastic contacts to activate, three things must happen *before* `CollisionPipeline` is created:

1. **Mesh shapes need `mesh.build_sdf()` called** before `add_shape_mesh()`. Newton's builder raises `ValueError` if `is_hydroelastic=True` but `mesh.sdf is None` (see [`builder.py:5228`](https://github.com/newton-physics/newton/blob/main/newton/_src/sim/builder.py#L5228)).

2. **Shapes need `ShapeConfig(is_hydroelastic=True)`** to get the `HYDROELASTIC` flag in `shape_flags`.

3. **Both shapes in a contact pair** must have the `HYDROELASTIC` flag. `HydroelasticSDF._from_model()` checks this and returns `None` if no hydroelastic pairs exist.

With PR #5219 alone, setting `sdf_hydroelastic_config=HydroelasticSDFCfg()` will **silently have no effect** — the pipeline is configured but no shapes have the flag, so `_from_model()` returns `None` and the pipeline falls back to point contacts. Users may believe hydroelastic is enabled when it isn't.

## Suggested Feedback

### 1. Add a warning when hydroelastic config has no effect

In `_initialize_contacts()`, after creating the pipeline, check if hydroelastic is actually active:

```python
if cls._collision_cfg is not None:
    cfg_dict = cls._collision_cfg.to_dict()
    hydro_cfg = cfg_dict.pop("sdf_hydroelastic_config", None)
    if hydro_cfg:
        cfg_dict["sdf_hydroelastic_config"] = HydroelasticSDF.Config(**hydro_cfg)
    cls._collision_pipeline = CollisionPipeline(cls._model, **cfg_dict)

    # Warn if hydroelastic was requested but no shapes qualify
    if hydro_cfg and cls._collision_pipeline.hydroelastic_sdf is None:
        logger.warning(
            "HydroelasticSDFCfg was set but no hydroelastic shape pairs were found. "
            "Ensure shapes have SDF built (mesh.build_sdf()) and is_hydroelastic=True "
            "in their ShapeConfig. Falling back to point contacts."
        )
```

### 2. Document the prerequisite in `HydroelasticSDFCfg` docstring

The current docstring says "enables hydroelastic contacts with the specified parameters" but doesn't mention that shapes must be prepared. Suggest adding:

```python
"""Configuration for SDF-based hydroelastic collision handling.

...

.. note::
    For hydroelastic contacts to be generated, shapes must have:

    1. SDF data built on their mesh (via ``mesh.build_sdf()``).
    2. The ``HYDROELASTIC`` flag set (via ``ShapeConfig(is_hydroelastic=True)``).

    Both shapes in a colliding pair must meet these requirements. If no qualifying
    shape pairs exist, this config has no effect and the pipeline uses point contacts.
"""
```

### 3. Consider a minimal SDF setup helper

Users of PR #5219 currently have no way to get hydroelastic working without writing custom code to iterate over builder shapes and call `mesh.build_sdf()`. A minimal helper could bridge this:

```python
@classmethod
def _ensure_sdf_on_meshes(cls, builder: ModelBuilder, sdf_max_resolution: int = 128):
    """Build SDF on mesh shapes that don't already have it."""
    from newton import GeoType, ShapeFlags
    for i in range(builder.shape_count):
        if builder.shape_type[i] == GeoType.MESH:
            mesh = builder.shape_source[i]
            if mesh is not None and mesh.sdf is None:
                mesh.build_sdf(max_resolution=sdf_max_resolution)
            builder.shape_flags[i] |= ShapeFlags.HYDROELASTIC
```

However, this overlaps significantly with PR #5160's `_apply_sdf_config()`. If PR #5160 is also expected to land, this may not be needed.

**Review by AntoineRichard — APPROVED** [review]
